source/ScreensaverHero.hx

```diff
@@ -3,84 +3,64 @@ package;
 import flixel.FlxG;
 import flixel.FlxSprite;
 import flixel.math.FlxRandom;
+import flixel.math.FlxVelocity;
+import flixel.util.FlxColor;
 import flixel.util.FlxSpriteUtil;
 
 class ScreensaverHero extends FlxSprite
 {
-	public var direction:Int;
+	public var direction:Float;
+	public var randomDirection:FlxRandom = new FlxRandom();
 
 	public function new(x:Int, y:Int)
 	{
 		super(x, y);
 		var randomDirection:FlxRandom = new FlxRandom();
-		this.direction = randomDirection.int(0, 7);
+		this.makeGraphic(25, 25, FlxColor.WHITE);
+		this.direction = randomDirection.float(0, 6.28318530718);
+		randDirectionMove(this.direction);
 	}
 
-	override function update(elapsed:Float)
+	public function randDirectionMove(direction)
+	{
+		FlxVelocity.accelerateFromAngle(this, direction, 25, 50, true);
+	}
+
+	public function colorChange(x:Float, y:Float)
 	{
-		if (this.x < 500)
+		if (x > y)
 		{
 			this.color = 0xff0000;
 		}
-		if (this.x > 500)
-		{
-			this.color = 0x00ff00;
-		}
-		if (this.direction == 0)
-		{
-			this.x--;
-		}
-		else if (this.direction == 1)
-		{
-			this.x--;
-			this.y--;
-		}
-		else if (this.direction == 2)
-		{
-			this.y--;
-		}
-		else if (this.direction == 3)
-		{
-			this.x++;
-		}
-		else if (this.direction == 4)
-		{
-			this.x++;
-			this.y++;
-		}
-		else if (this.direction == 5)
-		{
-			this.y++;
-		}
-		else if (this.direction == 6)
+		if (x < y)
 		{
-			this.x--;
-			this.y++;
+			this.color = 0x0000ff;
 		}
 		else
 		{
-			this.x++;
-			this.y--;
+			this.color = 0x00ff00;
 		}
+	}
+
+	override function update(elapsed:Float)
+	{
+		colorChange(this.x / FlxG.width, this.y / FlxG.height);
 		if (FlxG.mouse.pressed)
 		{
-			this.direction++;
-			if (this.direction == 8)
-			{
-				this.direction = 0;
-			}
+			var randomDirection:FlxRandom = new FlxRandom();
```

Comment:
Creating a new random number generator every time you need a number is overkill - I mentioned it in the last review, but declaring an instance of `FlxRandom` means you're setting up a generator that's meant to be used over and over again. 

It's most efficient to just use a single instance of `FlxRandom` for everything unless you have a specific need for sets of random numbers to be generated separately from one another.

As an example, maybe you want to make level generation use random numbers generated from an optional user-provided seed (so they can play or share that same level in the future), but you want random events inside the level to be unseeded so the gameplay stays unpredictable. 

In that sort of case, you could create a new `FlxRandom` instance when it's time to generate a level, passing the user-provided seed into its constructor. That `FlxRandom` would be used during all of the level generation code, while all other non-seeded random events could just rely on the `FlxRandom` statically set up by `FlxG`.

No change needed for this PR, but thought it might be worth elaboration